lib/wifi_user/use_case/email_signup.rb

```diff
@@ -46,6 +50,18 @@ def send_rejected_email_address_email(email_address)
     )
   end
 
+  def send_sponsor_signup_instructions(email_address)
+    sponsee = WifiUser::Repository::User.find(contact: email_address)
+    login_details = user_model.generate(contact: email_address)
+
+    send_email_gateway.execute(
```

Comment:
line 43 and line 32 don't inject an email gateway. Better stay consistent and it removes one more parameter / injection in the constructor